### ISSUE TITLE
Split building into 2 identical connected buildings

### ISSUE THREAD

**Issue by doublah** [openstreetmap/iD#6988]
Currently mapping semi-detached houses (which are very common in the UK and some other places) is a process which could be better optimized.

Currently if you wanted to add a semi-detached building, you'd use buildings drawn from satellite imagery (and this is probably the easiest way to add buildings), then add new nodes to the half way point, delete the other half of the building, then copy and reflect the building, then try to line it up to connect with the other house and attach the connecting nodes.

It's long, annoying for a lot of buildings, and inaccurate.
My solution to deal with this would be a split vertically/split horizontally option on the right click, similar to the existing reflect options.
This would make the process as simple as add house, square, split, repeat.

**Comment by bhousel** [comment]
Yes, occasionally we get requests for splitting like this..
We don't have this feature in iD, but you could use the [JOSM Terracer plugin](https://wiki.openstreetmap.org/wiki/JOSM/Plugins/Terracer) to achieve what you want..

**Comment by GenericError** [comment]
Unless anybody objects, I'd like to throw my hat in the ring and give this one a crack. I've jotted down some pseudo code and will keep the issue updated.

**Comment by Chaz6** [comment]
See also #6784 related to this.

**Comment by GenericError** [comment]
Would it be possible for the wip tag to be added to this issue? I've determined a working algorithm on paper, now I need to implement it in code.

**Comment by bhousel** [comment]
Ok, I added the label

**Comment by DaCor1** [comment]
Big +1 for this one. It will be especially useful for adding long terraces of houses

**Comment by tordans** [comment]
Recently, I made this building (https://www.openstreetmap.org/relation/241445) more precise based on official Data and though about this thread: Ideally, the splitting will also work with relations. Which will probably be tricky, since for some – like the one above – the relation would become obsolete but for others, like https://www.openstreetmap.org/relation/961130, (depending on the split) it would still be needed.

**Comment by bhousel** [comment]
Yeah as I said before, use the JOSM Terracer plugin if you want this now. We don’t have the capacity to build every thing into iD that people ask for.

**Comment by kymckay** [comment]
Hmm 🤔 @GenericError Is this still WIP for you?

Something I've considered implementing also, but always wondered what the best UX for such an operation would be.

It would be nice to also allow arbitrary splitting of an area using any drawn line across it (with the same operation) if the user selects the area and a way element.

Similarly, it would be nice to allow the user to split along the minor axis of a rectangle (rather than the typical major axis splitting).

**Comment by MASangys** [comment]
Truly support this feature!

**Comment by GenericError** [comment]
If somebody would like to take over the reigns of this then they are more than welcome to; essentially the closest I worked out would be to: draw a bounding box around the building (as seems to already be implemented as a function in iD), draw a line down the centre, create nodes at the intersection and reasign nodes in the way to create an additional building and resize the original building to only outline half of it

**Comment by nc011** [comment]
Ideally, this should be to _n_ number of identical buildings - not just two (but perhaps that's been implied already). Would be great to see implemented - it's something I would definitely make use of.

**Comment by k-yle** [comment]
For anyone following this issue: there's a prototype in #9020. Any suggestions for the UX would be great

**Comment by jidanni** [comment]
https://vespucci.io/help/en/Way%20selected/#split-in-to-polygons is great! Perhaps the same thing could be implemented in iD. The tags are retained. All I have to do is update the address number in the two new polygons.

**Comment by danieldegroot2** [comment]
See also #9334.
